pkg/util/ranger/detacher.go

```diff
@@ -462,6 +463,18 @@ func (d *rangeDetacher) detachCNFCondAndBuildRangeForIndex(conditions []expressi
 		// TODO: we will optimize it later.
 		res.RemainedConds = AppendConditionsIfNotExist(res.RemainedConds, remainedConds)
 		res.Ranges = ranges
+		if bestCNFItemRes != nil && res != nil && len(res.Ranges) != 0 {
```

Comment:
Could you please comment this case " (heuristics applied for long lists or we turn off the intersection)" in front of here?

Reply:
done